TorchSharp/FileOperations.cs

```diff
@@ -116,6 +114,69 @@ public long WriteTensor(TorchSharp.ByteTensor tensor)
             return THFile_writeByteRaw(this.handle, tensor.Data, tensor.NumElements);			
         }
 
+        [DllImport("caffe2")] 
+        extern static byte THFile_readCharScalar(HType self);
+
+        /// <summary>
+        ///   Read one byte from the file.
+        /// </summary>
+        /// <returns>A byte read from the current file position.</returns>
+        public byte ReadChar() { return THFile_readCharScalar(this.handle); }
+
+        [DllImport("caffe2")] 
+        extern static void THFile_writeCharScalar(HType self, byte scalar);
+
+        /// <summary>
+        ///   Write one byte to the file.
+        /// </summary>
+        /// <param name="value">A byte to write at the current file position.</param>
+        public void WriteChar(byte value) { THFile_writeCharScalar(this.handle, value); }
+
+        [DllImport("caffe2")] 
+        extern static long THFile_readCharRaw(HType self, IntPtr data, long n);
+        
+        /// <summary>
+        ///   Read bytes from the file into the given byte array.
+        /// </summary>
+        /// <param name="data">An array to place the data in after reading it from the file.</param>
+        /// <param name="n">The maximum number of bytes to read.</param>
+        /// <returns>The number of bytes read.</returns>
+        public long ReadChars(byte[] data, int n)
+        {
+            if (n > data.Length)
+                throw new ArgumentOutOfRangeException("n cannot be greater than data.Length");
+            unsafe
+            {
+                fixed (byte *dest = data)
+                {
+                    var readItems = THFile_readCharRaw(this.handle, (IntPtr)dest, n);
+                    return readItems;
+                }
+            }
+        }
+
+        [DllImport("caffe2")] 
+        extern static long THFile_writeCharRaw(HType self, IntPtr data, long n);
+
+        /// <summary>
+        ///   Write bytes to the file from the given byte array.
+        /// </summary>
+        /// <param name="data">An array containing data to be written to the file.</param>
+        /// <param name="n">The number of bytes to write. Pass -1 (default) to write the whole array.</param>
+        /// <returns>The number of bytes written.</returns>
+        public long WriteChars(byte[] data, int n = -1)
+        {
+            n = (n == -1) ? data.Length : Math.Min(n, data.Length);
```

Comment:
Should we have a check that n is not < -1?

Reply:
Good idea. I added it.